lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -971,6 +975,107 @@ def cmd_back(*args)
               driver.destack_dispatcher
             end
           end
+        
+          def cmd_favorite_help
+            print_line 'Usage: favorite [mod1 mod2 ...]'
+            print_line
+            print_line 'Add one or multiple modules to the list of favorite modules stored in the fav_modules file'
+            print_line 'If no module name is specified, the command will add the active module if there is one'
+            print_line
+            print_line 'OPTIONS:'
+            print_line '  -d                Delete the provided module(s) or the active module from the fav_modules file'
+            print_line '  -D                Delete the fav_modules file'
+          end
+
+          #
+          # Helper method for cmd_favorite that writes modules to the fav_modules_file
+          #
+          def cmd_favorite_process(mod, del)
+            favs_file = Msf::Config.fav_modules_file
+            if File.exists?(favs_file)
+              unless File.readable?(favs_file)
+                print_error("Unable to read #{favs_file}")
+                return false
+              end
+              
+              contents = File.read(favs_file)
+              if contents.include?(mod.fullname)
+                unless del
+                  print_warning("Module #{mod.fullname} has already been favorited and will not be added to #{favs_file}")
+                  return
+                end
+
+                # remove module from contents (deletion happens below)
+                contents = contents.split("\n")
```

Comment:
In order to play nice with Windows newlines, just using the default is fine here.
```suggestion
                contents = contents.split
```

Reply:
nice catch, done!